src/components/Products/ProductList/ProductList.component.jsx

```diff
@@ -27,19 +35,48 @@ const ProductList = () => {
     setPage,
     totalPages,
   } = usePagination({
-    contentPerPage: 16,
+    contentPerPage: 12,
     count: products.length,
   });
 
   // setting up filter logic
   const [filterArray, setFilterArray] = useState([]);
 
+  useEffect(() => {
+    if (slugCategory !== '') {
+      setFilterArray([...filterArray, slugCategory]);
+      console.log(filterArray);
+    }
+  }, []);
+
   const categoryFilter = (id) => {
     filterArray.includes(id)
       ? setFilterArray(filterArray.filter((x) => x !== id))
-      : setFilterArray([...filterArray, id]);
+      : //setFilterArray(filterArray.push(id));
+        setFilterArray(...filterArray, id);
+    console.log(filterArray);
+    const filtered = products.filter((product) =>
+      product.data.category.slug.includes(filterArray)
+    );
+    console.log(filterArray, filtered);
   };
 
+  if (isCategoriesLoading) {
+    return <Loading />;
+  }
+
+  if (!categories && !isCategoriesLoading) {
+    return <Error type="categories" />;
+  }
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  if (!products && !isLoading) {
+    return <Error type="products" />;
```

Comment:
same as above, you have 6 conditionals in these 4 ifs, it can be simplified to 4 conditionals. One way of knowing this is you're are re-checking a conditional, it would usually warn you that you could reduce your checks.